### PR TITLE
Move in message marshalling functions from 'envelopespec' package.

### PR DESCRIPTION
**Pull request by danilvpetrov**
#### What change does this introduce?

This PR moves the message marshalling functions from [`dogmatiq/envelopespec`](https://github.com/dogmatiq/envelopspec) package into this repository.  

#### What issues does this relate to?

Fixes https://github.com/dogmatiq/marshalkit/issues/40.

#### Why make this change?

This change is required to relocate message marshalling functions to the current repository from [`dogmatiq/envelopespec`](https://github.com/dogmatiq/envelopspec) package as they are not related to the envelope spec definitions, but to message marshalling. 

#### Is there anything you are unsure about?

No.

### CODE REVIEW

**Comment by danilvpetrov** [comment]
I had to slightly modify the imports in `envelope_test.go` file compared to what it was before in the [`dogmatiq/envelopespec`](https://github.com/dogmatiq/envelopespec) package as there is a conflict with the `marshalkit.Marshaler` type and `marhalkit/fixtures.Marshaler` variable in the tests.

**Comment by jmalloc** [comment]
Should we wait until `envelopespec` has been moved/merged/tagged in `interopspec` before merging this?

**Comment by danilvpetrov** [comment]
Yeah we probably should as the imports and refs would have to be changed.

File: envelope.go
Comment (koden-km) [inline envelope.go:69]:
```suggestion
// UnmarshalEnvelopeTime unmarshals t from its RFC-3339 representation, as
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:11]:
```suggestion
// MustMarshalMessageIntoEnvelope marshals a Dogma message into an envelopespec.Envelope.
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:31]:
```suggestion
// UnmarshalMessageFromEnvelope unmarshals a Dogma message from an envelopespec.Envelope
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:45]:
```suggestion
// representation, as used within envelopespec.Envelope.
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:54]:
```suggestion
// representation, as used within envelopespec.Envelope.
```

File: envelope.go
Comment (danilvpetrov) [inline envelope.go:69]:
```suggestion
// UnmarshalEnvelopeTime unmarshals t from its RFC-3339 representation, as
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:16]:
```suggestion
var _ = Describe("func MustMarshalMessageIntoEnvelope()", func() {
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:36]:
```suggestion
var _ = Describe("func UnmarshalMessageFromEnvelope()", func() {
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:60]:
```suggestion
// within envelopespec.Envelope.
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:50]:
```suggestion
var _ = Describe("func MustMarshalEnvelopeIdentity()", func() {
```

File: envelope.go
Comment (jmalloc) [inline envelope.go:70]:
```suggestion
// used within envelopespec.Envelope.
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:64]:
```suggestion
var _ = Describe("func UnmarshalEnvelopeIdentity()", func() {
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:79]:
```suggestion
var _ = Describe("func MustMarshalEnvelopeTime()", func() {
```

File: envelope_test.go
Comment (koden-km) [inline envelope_test.go:95]:
```suggestion
var _ = Describe("func UnmarshalEnvelopeTime()", func() {
```

**Review by jmalloc — CHANGES_REQUESTED** [review]

**Review by koden-km — CHANGES_REQUESTED** [review]

**Review by danilvpetrov — COMMENTED** [review]
Leaving a comment so I can commit my own suggestion 😄

**Review by jmalloc — APPROVED** [review]

**Comment by danilvpetrov** [comment]
Sorry @koden-km I am merging this PR in, LMK if you have some comments later on.

**Pull request merged**
